src/OneDrive.Sdk.Authentication.WinStore/OnlineIdAuthenticationProvider.cs

```diff
@@ -36,12 +36,14 @@ public class OnlineIdAuthenticationProvider : MsaAuthenticationProvider
         private const string onlineIdServiceTicketRequestType = "DELEGATION";
         private readonly int ticketExpirationTimeInMinutes = 60;
         private readonly OnlineIdAuthenticator authenticator;
+        private readonly CredentialPromptType credentialPromptType;
 
         public OnlineIdAuthenticationProvider(
-            string[] scopes)
+            string[] scopes, CredentialPromptType credentialPromptType = CredentialPromptType.PromptIfNeeded)
```

Comment:
Rather than directly using CredentialPromptType, make a new enum above. That makes it so that client code doesn't need to use the OnlineIdAuthenticator namespace. Also, we may want our own custom values in the future.